process_win.cpp

```diff
@@ -5,9 +5,35 @@
 
 Process::Data::Data(): id(0), handle(NULL) {}
 
+namespace {
+
+// Simple HANDLE wrapper to close it automatically from the destructor.
+class Handle {
+public:
+  Handle() : handle(INVALID_HANDLE_VALUE) { }
+  ~Handle() {
+    close();
+  }
+  void close() {
+    if (handle != INVALID_HANDLE_VALUE)
+      ::CloseHandle(handle);
+  }
+  HANDLE detach() {
+    HANDLE old_handle = handle;
+    handle = nullptr;
```

Comment:
Should this line not be `handle = INVALID_HANDLE_VALUE`? Or else CloseHandle will be called on nullptr? INVALID_HANDLE_VALUE corresponds to static_cast<size_t>(-1) if I'm not mistaken. 

And sorry for being way too late on this PR!